### ISSUE TITLE
Bug: HorizontalRuleNode breaks TableCell in some case

### ISSUE THREAD

**Issue by turnerguo** [facebook/lexical#6459]
Lexical version:

## Steps To Reproduce

1. insert a HorizontalRuleNode in table.
2. select above blank paragraph before HorizontalRuleNode and press delete key
3. the empty paragraph will deleted, keep press delete key the TableCell will deleted too, see the video on detail.

https://github.com/user-attachments/assets/decd08b6-0fe7-4897-9ce2-852f11d45e69

**Comment by etrepum** [comment]
This no longer reproduces in the same way, but there are still some edge cases here because once the divider is deleted you can enter text directly into the TableCellNode, it does not create a wrapping ParagraphNode.

**Comment by ivailop7** [comment]
> This no longer reproduces in the same way, but there are still some edge cases here because once the divider is deleted you can enter text directly into the TableCellNode, it does not create a wrapping ParagraphNode.

I'm no longer able to reproduce this issue, I see even after deleting the horizontal line, it does insert a paragraph. I guess you've added a transform in the big tables rewrite?

**Comment by etrepum** [comment]
yeah https://github.com/facebook/lexical/blob/main/packages/lexical-table/src/LexicalTablePluginHelpers.ts#L68-L76
